cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/connector/resource/AzureDatabaseResourceService.java

```diff
@@ -141,6 +149,34 @@
                 : deleteResourceGroup(resources, cloudContext, force, client, persistenceNotifier, stack);
     }
 
+    public void handleTransientDeployment(AuthenticatedContext authenticatedContext, List<CloudResource> resources) {
+        Optional<String> deploymentNameOpt = resources.stream()
+                .filter(cloudResource -> cloudResource.getType() == ResourceType.ARM_TEMPLATE)
+                .map(CloudResource::getName)
+                .findFirst();
+        Optional<String> resourceGroupNameOpt = resources.stream()
+                .filter(cloudResource -> cloudResource.getType() == ResourceType.AZURE_RESOURCE_GROUP)
+                .map(CloudResource::getName)
+                .findFirst();
+        AzureClient client = authenticatedContext.getParameter(AzureClient.class);
+        LOGGER.debug("Database template saved: {}, resource group saved: {}", deploymentNameOpt, resourceGroupNameOpt);
+        if (deploymentNameOpt.isPresent() && resourceGroupNameOpt.isPresent()) {
```

Comment:
The functionality of cancelling a deployment (by id or  by resourceGroup) and then getting the resources from it could be extracted to a separate class. This functionality is used at two other places in some variation.